gcc/rust/resolve/rust-ast-resolve-path.cc

```diff
@@ -88,6 +88,8 @@ ResolvePath::resolve_path (AST::PathInExpression *expr)
       // resolve any generic args
       if (segment.has_generic_args ())
 	{
+	  auto empty = CanonicalPath::create_empty ();
+	  ResolveGenericArgs::go (segment.get_generic_args (), empty, empty);
```

Comment:
I think ResolveGenericArgs::go should be the main entry point for all of this. I think it should disambiguate then call ResolveType on the arguments etc.